bitseq/sequence.go

```diff
@@ -273,12 +272,9 @@ func (h *Handle) set(ordinal uint32, any bool, release bool) (uint32, error) {
 			continue
 		}
 
-		// Unless unexpected error, save private copy to local copy
+		// Previous atomic push was succesfull. Save private copy to local copy
 		h.Lock()
 		defer h.Unlock()
-		if h.dbIndex != ci {
-			return ret, fmt.Errorf("unexected database index change")
-		}
```

Comment:
As you are checking for h.dbIndex against ci, this should NOT have failed. But, the reason it failed is because of `watchForChanges` callback. But anyways, this is a wrong check and must go. So that is good.

But, can you also remove the `h.dbIndex = nh.dbIndex` in the following code ?
I dont want this to override a valid dbIndex update due to a remote operation.
Lets make sure that the dbIndex is updated only from the watch thread and make it consistent.